pkg/apis/pipeline/v1beta1/pipelinerun_conversion.go

```diff
@@ -226,3 +377,41 @@ func deserializePipelineRunResources(meta *metav1.ObjectMeta, spec *PipelineRunS
 	}
 	return nil
 }
+
+func convertTaskRunsToChildReference(ctx context.Context, taskRuns map[string]*PipelineRunTaskRunStatus) []v1.ChildStatusReference {
+	csrs := []v1.ChildStatusReference{}
+	for name, status := range taskRuns {
+		csr := v1.ChildStatusReference{}
+		// TOOD: is apiVersion required here? and is it set to be `v1beta1`?
```

Comment:
Does it hold that `v1beta1` are always for TaskRunStatus type as a ChildStatusReference and `v1alpha1` for `RunStatus`?

Reply:
Ok, so for `Run`s, yeah, it should always be `v1alpha1`, since there's never going to be a `v1beta1` for `Run` - we won't be converting from `v1alpha1.Run` to `v1beta1.CustomRun`, so the `v1alpha1.Run` object will still exist. For `TaskRun`s, I think the `apiVersion` should maybe be based on what the `PipelineRun` reconciler is creating - if the `v1beta1.TaskRun` has already been created, which is the case for any `TaskRun` referenced in `ChildReferences`, that's how we should refer to it. It may be _stored_ as a `v1.TaskRun` via conversion, but, if I'm remembering correctly, we can still reference it as `v1beta1`?

Reply:
Thanks Andrew, that makes perfect sense to me